src/hooks/useSidebarOrderedReports.tsx

```diff
@@ -66,17 +69,108 @@ function SidebarOrderedReportsContextProvider({
     const derivedCurrentReportID = currentReportIDForTests ?? currentReportIDValue?.currentReportIDFromPath ?? currentReportIDValue?.currentReportID;
 
     const policyMemberAccountIDs = useMemo(() => getPolicyEmployeeListByIdWithoutCurrentUser(policies, undefined, accountID), [policies, accountID]);
+    const reportsToDisplayRef = useRef<ReportsToDisplayInLHN>({});
+    const prevBetas = usePrevious(betas);
+    const prevPriorityMode = usePrevious(priorityMode);
+
+    /**
+     * Find the reports that need to be updated in the LHN
+     */
+    const getUpdatedReports = useCallback(() => {
+        let reportsToUpdate: string[] = [];
+
+        if (betas !== prevBetas || priorityMode !== prevPriorityMode) {
+            reportsToUpdate = Object.keys(chatReports ?? {});
+        } else if (reportUpdates) {
+            reportsToUpdate = Object.keys(reportUpdates ?? {});
+        } else if (reportNameValuePairsUpdates) {
+            reportsToUpdate = Object.keys(reportNameValuePairsUpdates ?? {}).map((key) => key.replace(ONYXKEYS.COLLECTION.REPORT_NAME_VALUE_PAIRS, ONYXKEYS.COLLECTION.REPORT));
+        } else if (transactionsUpdates) {
+            reportsToUpdate = Object.values(transactionsUpdates ?? {}).map((transaction) => `${ONYXKEYS.COLLECTION.REPORT}${transaction?.reportID}`);
+        } else if (transactionViolationsUpdates) {
+            reportsToUpdate = Object.keys(transactionViolationsUpdates ?? {})
+                .map((key) => key.replace(ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS, ONYXKEYS.COLLECTION.TRANSACTION))
+                .map((key) => `${ONYXKEYS.COLLECTION.REPORT}${transactions?.[key]?.reportID}`);
+        } else if (reportsDraftsUpdates) {
+            reportsToUpdate = Object.keys(reportsDraftsUpdates).map((key) => key.replace(ONYXKEYS.COLLECTION.REPORT_DRAFT_COMMENT, ONYXKEYS.COLLECTION.REPORT));
+        } else if (policiesUpdates) {
+            const updatedPolicies = Object.keys(policiesUpdates).map((key) => key.replace(ONYXKEYS.COLLECTION.POLICY, ''));
+            reportsToUpdate = Object.entries(chatReports ?? {})
+                .filter(([, value]) => {
+                    if (!value?.policyID) {
+                        return;
+                    }
+
+                    return updatedPolicies.includes(value?.policyID);
+                })
+                .map(([key]) => key);
+        }
+
+        return reportsToUpdate;
+        // eslint-disable-next-line react-compiler/react-compiler, react-hooks/exhaustive-deps
```

Comment:
Can you please think of a way to avoid this eslint disable? if not possible, leave a clear comment why its not possible